File: src/main/java/ac/dnd/server/admission/infrastructure/persistence/mapper/ApplicantPersistenceMapper.java
package ac.dnd.server.admission.infrastructure.persistence.mapper;

import org.springframework.stereotype.Component;

import ac.dnd.server.admission.domain.model.ApplicantData;
import ac.dnd.server.admission.domain.model.EventData;
import ac.dnd.server.admission.infrastructure.persistence.entity.Applicant;
import ac.dnd.server.admission.infrastructure.persistence.entity.Event;
import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class ApplicantPersistenceMapper {

	public Event eventDataToEntity(
		final EventData domain
	) {
		return new Event(
			domain.name(),
			domain.period()
		);
	}

	public EventData eventEntityToDomain(final Event entity) {
		return new EventData(
			entity.getId(),
			entity.getName(),
			entity.getPeriod()
		);
	}

	public ApplicantData applicantEntityToDomain(
		final Applicant entity
	) {
		return new ApplicantData(
			entity.getId(),
			entity.getEvent().getName(),
			entity.getName(),
			entity.getEmail(),
			entity.getType(),
			entity.getStatus(),
			entity.getEvent().getPeriod()
		);
	}
}

File: src/main/kotlin/ac/dnd/server/admission/infrastructure/persistence/mapper/ApplicantPersistenceMapper.kt
package ac.dnd.server.admission.infrastructure.persistence.mapper

import ac.dnd.server.admission.domain.model.ApplicantData
import ac.dnd.server.admission.domain.model.EventData
import ac.dnd.server.admission.infrastructure.persistence.entity.ApplicantEntity
import ac.dnd.server.admission.infrastructure.persistence.entity.EventEntity
import org.springframework.stereotype.Component

@Component
class ApplicantPersistenceMapper {

    fun eventDataToEntity(domain: EventData): EventEntity {
        return EventEntity(
            domain.name,
            domain.period,
            domain.resultAnnouncementDateTime,
            domain.status
        )
    }

    fun eventEntityToDomain(entity: EventEntity): EventData {
        return EventData(
            entity.id!!,
            entity.name,
            entity.period,
            entity.resultAnnouncementDateTime,
            entity.status
        )
    }

    fun applicantEntityToDomain(entity: ApplicantEntity): ApplicantData {
        return ApplicantData(
            entity.id!!,
            entity.event!!.name,
            entity.name,
            entity.email,
            entity.type,
            entity.status,
            entity.event!!.period
        )
    }
}
